domain/src/validation.js

```diff
@@ -71,6 +128,50 @@ function checkChar(char) {
     };
 }
 
+// > checkSignUp({email: 'userEmail', name: 'userName', password: 'userPass'})
+// {isValid: true, errors: [], user : {email: 'userEmail', name: 'userName', password: 'userPass'}}
+function checkSignUp(user) {
+    user = _.pick(user, ['email', 'name', 'password']);
+    let errors = [];
+    let notString = filterNotString(user, ['email', 'name', 'password']);
+    if (notString.length) {
+        errors.push({
+            id: "mustBeString",
+            properties: notString
+        });
+    }
```

Comment:
Wow, I must have been soooooooooooooooooooooooooooo drunk when we designed validation process :roll_eyes: 
We have, let's say, special nontraditional approach here - every other person in the world does validation property by property. And we do validation rule by rule. That is really fucked up in terms of reuse - if we have certain complex validation rules for a certain field, we can't simply reuse it in another model.

I guess it's too late to change it now, but just keep in mind that this is bad, really bad, never do that again. Always validate revolving around properties, not around validation rules.

P. S. thats how hipsters are born ...